lib/mix/lib/mix/compilers/erlang.ex

```diff
@@ -22,7 +22,7 @@ defmodule Mix.Compilers.Erlang do
 
       compile manifest, [{"src", dest}], :lfe, :beam, opts, fn input, output ->
         :lfe_comp.file(to_erl_file(input),
-                       [{output_dir, Path.dirname(output)}, :return])
+                       [{:outdir, Path.dirname(output)}, :return, :report])
```

Comment:
If you pass :report, the errors will be reported twice, because once they are returned we printed them too, no?

Reply:
Not really, in the beginning I tried without `:report` and I was not seeing them at all, that's why I added the `:report` bit and started seeing them. In the moment, when I compile file with bad syntax the output is OK:

```
> mix compile
Compiling 1 file (.lfe)
src/echo.lfe:11: unknown form
src/echo.lfe:9999: unbound function: #(start 2)
```

Reply:
@meddle0x53 ideally we want to return them because then you get better tooling, such as editor support. Can you double check that we are returning warnings and errors with the `:return` flag and without the `:report` one?

Reply:
Also both `:return` and `:report` are passed to the `Erlang` compiler when the `compile.erlang` task is ran (that's where I saw passing the `:report` bit and tried it): https://github.com/elixir-lang/elixir/blob/1c5ed3840d608b4488477c52b549c494409e47d0/lib/mix/lib/mix/tasks/compile.erlang.ex#L79